### PR TITLE
Pull Request automation in krel release-notes

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?

 /kind feature

#### What this PR does / why we need it:

This PR automates the creation of pull requests to update the release notes draft in `k/sig-release` and the release notes website in `ksigs/release-notes`. It mainly puts to work changes to the github library merged in #1287 and #1268 so most of the work was already in place.

It also verifies that the repositories used by the user are actual forks of their kubernetes counterparts..

#### Which issue(s) this PR fixes:

Related to #834

#### Special notes for your reviewer:

I've broken the PR into two commits, one for the automation of the release notes draft and one for the website patches.

The website portion is a bit temporary and will be removed soon as we are about to begin implementing some changes to move the json files to a GCP bucket. I plan to refactor some of the release notes subcommand code before that.

#### Does this PR introduce a user-facing change?

After this patch, running krel with `--create-draft-pr` or `--create-website-pr` will now _really_ create pull requests in github.

```release-note
* Automate the creation of pull requests to update the release notes draft in `kubernetes/sig-release`
* Automate the creation of pull requests in `kubernetes-sigs/release-notes` to update relnotes.k8s.io 
```

### CODE REVIEW

**Comment by puerco** [comment]
For reference, kubernetes-sigs/release-notes#153 was automatically generated using these changes.

File: cmd/krel/cmd/release_notes.go [hunk 1]
```diff
@@ -48,6 +48,10 @@ const (
 	defaultKubernetesSigsOrg = "kubernetes-sigs"
 	// defaultKubernetesSigsRepo relnotes.k8s.io repository name
 	defaultKubernetesSigsRepo = "release-notes"
+	// defaultKubernetesSigReleaseOrg org owner of the sig-release repo
+	defaultKubernetesSigReleaseOrg = "kubernetes"
+	// defaultKubernetesSigReleaseRepo name of the sig-release repository
+	defaultKubernetesSigReleaseRepo = "sig-release"
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:54]: We could avoid that new const when using `git.DefaultGithubReleaseRepo`.

File: cmd/krel/cmd/release_notes.go [hunk 2]
```diff
@@ -272,9 +284,69 @@ func runReleaseNotes() (err error) {
...
+	return nil
+}
+
 // validateDraftPROptions checks if we have all needed parameters to create the Release Notes PR
 func validateDraftPROptions() error {
 	if releaseNotesOpts.createDraftPR {
+		token, isset := os.LookupEnv("GITHUB_TOKEN")
+		if !isset || token == "" {
+			return errors.New("Cannot create release notes draft if GITHUB_TOKEN is not set")
+		}
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:348]: Hm, either we add it as public utility function to `pkg/github/github.go` or we just use `github.TokenEnvKey`   here.

File: cmd/krel/cmd/release_notes.go [hunk 3]
```diff
@@ -301,6 +373,15 @@ func validateDraftPROptions() error {
 // validateWebsitePROptions checks if we have all needed parameters to create the Release Notes PR
 func validateWebsitePROptions() error {
 	if releaseNotesOpts.createWebsitePR {
+		token, isset := os.LookupEnv("GITHUB_TOKEN")
+		if !isset || token == "" {
+			return errors.New("Cannot create release notes json file if GITHUB_TOKEN is not set")
+		}
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:379]: The same for this as mentioned above.

**Comment by puerco** [comment]
I had forgotten to cc the team
/cc @reylejano-rxm @JamesLaverack @AGMETEOR @wilsonehusin

File: cmd/krel/cmd/release_notes.go (hunk 1 above)
Comment (puerco) [reply cmd/krel/cmd/release_notes.go:54]: Fixed. Also got rid of `defaultKubernetesSigReleaseOrg` using `git.DefaultGithubOrg`

File: cmd/krel/cmd/release_notes.go (hunk 2 above)
Comment (puerco) [reply cmd/krel/cmd/release_notes.go:348]: Switched to `github.TokenEnvKey` although i think we should add a new function to ensure an authenticated client. I'll add it tomorrow.

File: cmd/krel/cmd/release_notes.go (hunk 3 above)
Comment (puerco) [reply cmd/krel/cmd/release_notes.go:379]: Same, using `github.TokenEnvKey`

File: cmd/krel/cmd/release_notes.go (hunk 1 above)
Comment (saschagrunert) [reply cmd/krel/cmd/release_notes.go]: Awesome, thank you!

**Comment by puerco** [comment]
I'm merging this one and I'll start fixing the issue found in kubernetes-sigs/release-notes#153 
/hold cancel

**Pull request merged**